torch/_subclasses/fake_tensor.py

```diff
@@ -320,6 +322,79 @@ def mk_fake_tensor(make_meta_t):
         )
         if out is NotImplemented:
             raise UnsupportedFakeTensorException("meta converter nyi")
+
+        from torch._dynamo.source import RandomValueSource
+
+        # TODO: extend this for smaller dtypes too.  Some care will have to be
+        # taken though, because smaller dtypes will compute differently but
+        # Python sympy compute is always done at arbitrary precision.
+        value = None
+        if (
+            not self.export
+            and type(t) is torch.Tensor
+            # TODO: It sure would be nice to test if something was a plain
+            # tensor without having to enumerate all the cases
+            and t.layout == torch.strided
+            and not (
+                t.is_sparse
+                or t.is_nested
+                or is_functorch_wrapped_tensor(t)
+                or is_legacy_batchedtensor(t)
+                or torch._is_functional_tensor(t)
+            )
+            and t.dim() == 0
+            and t.device.type == "cpu"
+            and t.dtype in [torch.int64, torch.float64]
```

Comment:
sorry, why does int64 work but not int32 ? Neither exactly models python ints since python will do big ints

Reply:
int32 is just more likely to overflow. int64 is the best we can do, seems safest. I do think in principle we can extend this to all the small dtypes, it's just not necessary for what I want it for (float inputs to graph; don't even need int64).

Reply:
int32 should also be valid (and all signed types) as overflow is UB.

Reply:
could we add runtime asserts for the int32 dtypes that they are within bounds ?

Reply:
I mean, if the original PyTorch program was relying on this UB, chances are it's already going to be buggy no matter what we do, as gcc will certainly assume so, so we might as well go all the way and assume it here and perform this transformation.